lib/mongo/operation/shared/read_preference_supported.rb

```diff
@@ -95,14 +92,18 @@ def command(connection)
       #   operation will be executed on.
       #
       # @return [ Hash ] New command document to send to the server.
-      def update_selector_for_read_pref(sel, connection)
+      def add_read_preference_legacy(sel, connection)
         if read && connection.description.mongos? && read_pref = read.to_mongos
-          Mongo::Lint.validate_camel_case_read_preference(read_pref)
-          sel = sel[:$query] ? sel : {:$query => sel}
-          sel = sel.merge(:$readPreference => read_pref)
-        else
-          sel
+          # If the read preference contains only mode and mode is secondary
+          # preferred and we are sending to a pre-OP_MSG server, this read
+          # preference is indicated by the :slave_ok wire protocol flag.
```

Comment:
Took me a moment to realise that the comment explains when the `$readPreference` option doesn't need to be added. Maybe add another sentence like "In all other cases, add the `$readPreference` option" might help clarify this.

Reply:
Sure, I added a note to this effect.